src/openrct2/management/research.c

```diff
@@ -643,3 +643,14 @@ void reset_researched_scenery_items()
 		gResearchedSceneryItems[i] = 0xFFFFFFFF;
 	}
 }
+
+void reset_researched_ride_types_and_entries()
+{
+	for (sint32 i = 0; i < 4; i++) {
```

Comment:
Not sure why this uses 4, when the maximum number of researched ride types is actually 8....